platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManager.java

```diff
@@ -0,0 +1,155 @@
+    static final String FELIX_FILENAME = "felix.fileinstall.filename";
+
+    static {
+        EXCLUDED_PROPERTIES.add(SERVICE_PID);
+        EXCLUDED_PROPERTIES.add(SERVICE_FACTORYPID);
+        EXCLUDED_PROPERTIES.add(FELIX_FILENAME);
+    }
+
+    private final EncryptionAgent agent;
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager) {
+        this(persistenceManager,
+                System.getProperty("ddf.home")
+                        .concat("/etc/certs"));
+    }
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager,
+            String passwordDirectory) {
+        super(persistenceManager);
+        this.agent = new EncryptionAgent(passwordDirectory);
+    }
+
+    @Override
+    public Dictionary load(String pid) throws IOException {
+        return decryptAndReturnDictonary(super.load(pid));
+    }
+
+    @Override
+    public void store(String pid, Dictionary properties) throws IOException {
+        super.store(pid, encryptAndReturnDictonary(properties));
+    }
+
+    // The 2.3.7 version of the maven-bundle-plugin fails on lambdas,
+    // so we need to use Java 7 constructs
+    private Dictionary<String, Object> encryptAndReturnDictonary(Dictionary<String, Object> dict) {
+        Dictionary<String, Object> encryptedDictionaryToReturn = new Hashtable<>();
+        Enumeration<String> keys = dict.keys();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            encryptedDictionaryToReturn.put(key, encryptString(key, dict.get(key)));
+        }
+        return encryptedDictionaryToReturn;
+    }
+
+    // The 2.3.7 version of the maven-bundle-plugin fails on lambdas,
+    // so we need to use Java 7 constructs
+    private Dictionary<String, Object> decryptAndReturnDictonary(Dictionary<String, Object> dict) {
+        Dictionary<String, Object> decryptedDictionaryToReturn = new Hashtable<>();
+        Enumeration<String> keys = dict.keys();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            decryptedDictionaryToReturn.put(key, decryptString(key, dict.get(key)));
+        }
+        return decryptedDictionaryToReturn;
+    }
+
+    private Object encryptString(String key, Object value) {
+        if (value instanceof String && !EXCLUDED_PROPERTIES.contains(key)) {
+            return agent.encrypt((String) value);
+        }
+        return value;
+    }
+
+    private Object decryptString(String key, Object value) {
+        if (value instanceof String && !EXCLUDED_PROPERTIES.contains(key)) {
+            return agent.decrypt((String) value);
+        }
+        return value;
+    }
+
+    private static class EncryptionAgent {
+        private Crypter crypter;
+
+        public EncryptionAgent(String passwordDirectory) {
+            synchronized (EncryptionAgent.class) {
+                if (!new File(passwordDirectory.concat("/bundlecache")).exists()) {
```

Comment:
❗️ How about Windows?

Reply:
It will be fine. Encryption service has been doing this forever. The File ctor is protecting us.